backend/src/main/java/com/google/sps/servlets/RemoveFavouriteServlet.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.utils.UserCollector;
+import com.google.sps.utils.UserConstants;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/api/remove-favourite")
+public class RemoveFavouriteServlet extends AuthenticationServlet {
+  @Override
+  protected void get(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // no get request
+  }
+
+  /** Removes a recipe from user's list of favourite recipes */
+  @Override
+  protected void post(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Do you need to do any sanitization here? Is it possible for example for the user to tap remove twice quickly and this will be called again with the ID of a recipe that was already removed? If so, what happens? Is it ignored or could it crash?

Reply:
`favourites.remove(recipeId)` returns true/false depending on whether the recipe was deleted or not. It can throw a NullPointerException if favourites is null, so I added a null check for it. If it's null, I initialise a new array list and still try to remove the recipe from it. In this case, it is possible for the user to tap remove twice and the app won't crash.

Reply:
SGTM